sky/packages/sky/lib/src/material/dropdown.dart

```diff
@@ -0,0 +1,258 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'icon.dart';
+import 'ink_well.dart';
+import 'shadows.dart';
+import 'theme.dart';
+
+const Duration _kMenuDuration = const Duration(milliseconds: 300);
+const double _kMenuItemHeight = 48.0;
+const double _kMenuHorizontalPadding = 36.0;
+const double _kBaselineOffsetFromBottom = 20.0;
+const Border _kDropdownUnderline = const Border(bottom: const BorderSide(color: const Color(0xFFBDBDBD), width: 2.0));
+
+class _DropdownMenu extends StatelessComponent {
+  _DropdownMenu({
+    Key key,
+    this.items,
+    this.rect,
+    this.performance,
+    this.selectedIndex,
+    this.level: 4
+  }) : super(key: key) {
+    assert(items != null);
+    assert(performance != null);
+  }
+
+  final List<DropdownMenuItem> items;
+  final Rect rect;
+  final PerformanceView performance;
+  final int selectedIndex;
+  final int level;
+
+  Widget build(BuildContext context) {
+    // The menu is shown in three stages (unit timing in brackets):
+    // [0 - 0.25] - Fade in a rect-sized menu container with the selected item.
+    // [0.25 - 0.5] - Grow the otherwise empty menu container from the center until it's
+    //   big enough for as many items as we're going to show.
+    // [0.5 - 1.0] Fade in the remaining visible items from top to bottom.
+    //
+    // When the menu is dismissed we do our best to just fade the entire thing out
```

Comment:
It's guaranteed to fade out. The doing my best bit was about not reversing the menu resize or individual item fades that occur when the menu is popped up.  So I guess it's just "humble speak". I'll remove the humility :-).